modules/auxiliary/scanner/dcerpc/petitpotam.rb

```diff
@@ -78,19 +80,32 @@ def run_host(_ip)
     rescue Rex::Proto::SMB::Exceptions::Error, RubySMB::Error::RubySMBError => e
       fail_with(Failure::NoAccess, "Unable to authenticate ([#{e.class}] #{e}).")
     end
+    report_service(service_data)
+
+    begin
+      tree = simple.client.tree_connect("\\\\#{sock.peerhost}\\IPC$")
+    rescue RubySMB::Error::RubySMBError => e
+      raise StandardError, "Unable to connect to the remote IPC$ share ([#{e.class}] #{e})."
+    end
 
     handle_args = PIPE_HANDLES[datastore['PIPE'].to_sym]
     fail_with(Failure::BadConfig, "Invalid pipe: #{datastore['PIPE']}") unless handle_args
 
-    @handle = dcerpc_handle(
-      handle_args[:uuid],
+    # rename tree_file
+    @pipe = tree.open_file(filename: handle_args[:filename], write: true, read: true)
+    handle = dcerpc_handle(
```

Comment:
*Technically* this handle isn't used for anything but printing stats information anymore. IMHO the verbosity with which it prints the endpoint and the UUID is extremely helpful in this scenario because the nature of the attack uses uncommon UUID and endpoint combinations. For that reason, I opted to keep this output.